server/routes/createGoal/createGoalController.ts

```diff
@@ -66,13 +66,24 @@ export default class CreateGoalController {
     const { prisonNumber } = req.params
     const targetDate = parseDate(req, 'targetDate')
     req.session.addStepForm = { ...req.body, targetDate }
+    const { addStepForm } = req.session
 
-    const errors = validateAddStepForm(req.session.addStepForm)
+    const errors = validateAddStepForm(addStepForm)
     if (errors.length > 0) {
       req.flash('errors', errors)
       return res.redirect(`/plan/${prisonNumber}/goals/add-step`)
     }
 
+    // Redirect to the desired page based on the form action
+    if (addStepForm.action === 'add-another-step') {
+      // The next PR will work out where to store steps as each is submitted
+
+      // Initialize a new AddStepForm with the next step number
+      const nextStepNumber = Number(addStepForm.stepNumber) + 1
```

Comment:
Why not just `addStepForm.stepNumber + 1`? 🤔

Reply:
Excellent question 😁 - see my screenshot above which shows "Step 1111"
Even though our form type (in `@types`) defines the field as a `number`, at the point of it being rendered in the markup in the various `input` fields etc, and then subsequently submitted back in the HTTP request, all the fields become `string`s ... which is annoying !
It's because `request.params` (which is the map of form fields that are submitted on the POST request) is typed as a `Map<String, String>` (in java speak) 
When the controller constructs the form object from the request params (ie. `req.session.addStepForm = { ...req.body, targetDate }`) it annoying doesnt coerce the data types into the target object. It changes the types of the target object !!!! (because at runtime its all javascript, and js doesnt have type safety in the same way that typescript does. All the neat typescript stuff with types etc is all compile time and developer convenience. At runtime its all plain old javascript!)

So .... to answer your question ... at runtime at this line 82, `addStepForm.stepNumber` is a string, so `addStepForm.stepNumber + 1` concatenates it as a string (hence `Step 1111` in my screenshot!)

Reply:
I miss Kotlin! 😄